apps/web/lib/getMainDomainOrgRedirect.ts

```diff
@@ -0,0 +1,33 @@
+import type { IncomingMessage } from "http";
+
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+
+const log = logger.getSubLogger({ prefix: ["lib", "getMainDomainOrgRedirect"] });
+export const getMainDomainOrgRedirect = (req: IncomingMessage | undefined) => {
```

Comment:
If it needs just the URL, then it should just accept the URL. That is a clean code guideline.

Reply:
Also wouldn't it be better to move it to a method inside `packages/lib/server/repository/organization.ts`

Reply:
I think we shouldn't move it inside repository. Repository is supposed to be a DB layer. Application logic should remain separate IMHO.

But yeah I have thought about it multiple times that where should we hold all utility methods related to an entity. Maybe we could have lib/organization directory?

@zomars what do you think?

Reply:
if we are using `organizationSettings.orgAutoAcceptEmail` then this `getMainDomainOrgRedirect.ts` is not required.
@hariombalhara please clarify does `orgAutoAcceptEmail` store only hostname always and not the email id, else validation needs to added to extract hostname from email id, what i could see in code is , it is inferred to hold hostname only, because of the naming i got this query.

Reply:
Yeah, it is hostname only. The naming isn't clear. That Email in it actually means that all emails matching that domain will be automatically added to the organization.

Reply:
ok sure thanks for clarifying